### ISSUE TITLE
Add item tag / itemstack NBT support for catalyst items

### ISSUE THREAD

**Issue by Warbringer12** [CompactMods/CompactCrafting#35]
hi I would like the ability to use tags in the catalyst portion

**Comment by robotgryphon** [comment]
Per discord discussions, using `tag` on the catalyst definition makes the recipe show the tag data in game, but the NBT data (such as enchantments) is not respected during item comparison. To be evaluated next release.

**Comment by 854342681** [comment]
I use follow code to make a enchanted diamond sword as crystal, but it's not work.
```
{
  "type": "compactcrafting:miniaturization",
  "recipeSize": 3,
  "layers": [
     {
      "type":"compactcrafting:filled",
      "component": "s"
    },
    {
      "type": "compactcrafting:mixed",
      "pattern":[
        ["s","a","s"],
        ["s","e","s"],
        ["s","a","s"]
      ]
    },
    {
      "type":"compactcrafting:filled",
      "component": "s"
    }
  ],
  "catalyst":{
    "id":"minecraft:diamond_sword",
    "tag":"{Enchantments:[{\"id\":\"minecraft:sharpness\",lvl:5}]}",
    "Count":1
  },
  "components":{
    "s":{
      "type": "compactcrafting:block",
      "block": "pneumaticcraft:plastic_brick_lime"
    },
    "a":{
      "type": "compactcrafting:block",
      "block": "mekanism:induction_port"
    },
    "e":{
      "type": "compactcrafting:block",
      "block": "mekanism:elite_energy_cube"
    }
  },
  "outputs":[{
    "id":"masterfulmachinery:dragon_dra_energy_port_energy_input",
    "Count":1
  }]
}
```
and I also try change `"tag":"{Enchantments:[{\"id\":\"minecraft:sharpness\",lvl:5}]}",` to JSON object like this 
```
"tag":{
      "Enchantments":[{
        "id":"minecraft:sharpness",
        "lvl":5
      }]
    },
```
it can see in JEI, but normal diamond sword also could as crystal.

**Comment by 854342681** [comment]
Additionally, item with NBT can be regarded as crystal.
```
"outputs":[{
    "id":"minecraft:diamond_sword",
    "tag":{
      "Enchantments":[{
        "id":"minecraft:sharpness",
        "lvl":5
      }]
    },
    "Count":1
  }]
```

**Comment by robotgryphon** [comment]
Closed with the release of CC 1.0. Grab it on Curseforge, it's available!
